pkg/doc/signature/proof/proof_test.go

```diff
@@ -41,6 +41,22 @@ func TestProof(t *testing.T) {
 	require.Equal(t, "abc.com", p.Domain)
 	require.Equal(t, []byte(""), p.Nonce)
 	require.Equal(t, proofValueBytes, p.ProofValue)
+
+	// test created time with milliseconds section
+	p, err = NewProof(map[string]interface{}{
+		"type":               "type",
+		"creator":            "didID",
+		"verificationMethod": "did:example:123456#key1",
+		"created":            "2018-03-15T00:00:00.972Z",
+		"domain":             "abc.com",
+		"nonce":              "",
+		"proofValue":         proofValueBase64,
+	})
+	require.NoError(t, err)
+
+	created, err = time.Parse(time.RFC3339, "2018-03-15T00:00:00.972Z")
```

Comment:
how come this one isnt parsed with Nano?

Reply:
I forgot to change this to the `time.RFC3339Nano`... But it still works, also shocking for me :)
Btw json.Unmarshal(*time.Time) will also keep millisecond precision.